pkg/version/version.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/blang/semver"
+	log "github.com/sirupsen/logrus"
+)
+
+// providerVersion holds current provider version
+type providerVersion struct {
+	Version string `json:"version"`
+	Date    string `json:"date"`
```

Comment:
What is `Date`? how will it be used and who will use it?

Reply:
`Date` here is the build date and time for the provider image. This can be used in debug logs to see the build date is latest if the same image tag is rebuilt in some corner scenario.